txpipe/twopoint_null_tests.py

```diff
@@ -766,3 +767,272 @@ def write_output(self, source_list, lens_list, meta, results):
 
         # Finally, save the output to Sacc file
         S.save_fits(self.get_output("aperture_mass_data"), overwrite=True)
+
+
+
+class TXpureB(PipelineStage):
+    '''
+    Make shear B-mode measurements
+    '''
+    name     = "TXpureB"
+    parallel = False
+
+    inputs  = [
+               ("shear_photoz_stack"    , QPNOfZFile), 
+               ("source_maps"           , MapsFile),
+               ("mask"                  , MapsFile),
+               ("twopoint_data_real_raw", SACCFile),
+              ]
+    
+    outputs = [("twopoint_data_fourier_pureB", SACCFile)]
+
+    config_options = {
+                      'method'    : 'hybrideb', 
+                      'Nell'      : 20,         
+                      'lmin'      : 200,         
+                      'lmax'      : 2000,       
+                      'lspacing'  : 'log',       
+                      'bin_file'  : "",              
+                      'theta_min' : 2.5,      
+                      'theta_max' : 250,      
+                      'Ntheta'    : 1000,     
+                      'Nsim'      : 1000
+                      }
+    
+    def run(self):
+        import os
+        import pymaster as nmt
+        import healpy as hp
+        import sacc
+        import pyccl
+        import hybrideb
+        import datetime
+        import pickle
+        from tqdm import tqdm
+
+        if self.config["method"] == 'namaster_pureB' or self.config["method"] == 'namaster_nopureB':
```

Comment:
Actually the only difference between the two methods is the `purify_b` flag so if we try to separate the two methods, it would look identical except for the flag. Maybe I'm misunderstanding your suggestion.